auto-notify.plugin.zsh

```diff
@@ -57,12 +57,24 @@ function _auto_notify_message() {
             transient=""
         fi
         notify-send "$title" "$body" --app-name=zsh $transient "--urgency=$urgency" "--expire-time=$AUTO_NOTIFY_EXPIRE_TIME"
+        if [[ -n "$AUTO_NOTIFY_SOUND" ]]; then
+            paplay "$AUTO_NOTIFY_SOUND"
+        fi
+
     elif [[ "$platform" == "Darwin" ]]; then
-        osascript \
-          -e 'on run argv' \
-          -e 'display notification (item 1 of argv) with title (item 2 of argv)' \
-          -e 'end run' \
-          "$body" "$title"
+        if [[ -z "$AUTO_NOTIFY_SOUND" ]]; then
+            osascript \
+              -e 'on run argv' \
+              -e 'display notification (item 1 of argv) with title (item 2 of argv)' \
+              -e 'end run' \
+              "$body" "$title"
+        else
+            osascript \
+              -e 'on run argv' \
+              -e 'display notification (item 1 of argv) with title (item 2 of argv) sound name (item 3 of argv)' \
+              -e 'end run' \
+              "$body" "$title" "$AUTO_NOTIFY_SOUND"
+        fi
```

Comment:
Not too well-versed with shell scripting.  
But is it possible to extract the common command line arguments here out?  

Something like?  
```suggestion
        notification_body := "$body" "$title"
        if [[ ! -z "$AUTO_NOTIFY_SOUND" ]]; then
            notification_body += ("$AUTO_NOTIFY_SOUND")
        else

         osascript \
              -e 'on run argv' \
              -e 'display notification (item 1 of argv) with title (item 2 of argv) sound name (item 3 of argv)' \ (OPTIONAL CONCAT HERE AS WELL)
              -e 'end run' \
              ${notification_body[@]}
```

Reply:
Thank you for suggestion. I'm not sure if i get it right but made changes. Check it out please!